app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
```

Comment:
Ok, I see what you;re saying. When I started work on this PR it seemed like the choice of modals was determined by either LOA3 or LOA1 requests. Actually they are deetermined bya  user signing up or attempting to verify their identity with a currently existing user. I was planning on changing these names in a follow up PR.

You are right that the method to determine whether or not the user is in the LOA3 context isn't tied to whether or not they have a recovery code.

However, whether or not the user is attempting a sign up via LOA3 or LOA1, canceling at any point before the final verification screen is going to delete their account